### PR TITLE
implement basic RSS notifier for RPi locator

### PR DESCRIPTION
**Pull request by fisher60**

### CODE REVIEW

File: bot/cogs/rss.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            rss_data = feedparser.parse(await self.pull_rss_feed(RPI_RSS_FEED_URL))["entries"][:10]
+
+            for user_id in notifications:
+                user_notifs = notifications[user_id]
+                for notif in user_notifs.values():
+                    ping_member_data = data_to_ping_from_rss(notif.search_term, rss_data)
+                    if ping_member_data and not notif.notified:
+                        notif.notified = True
+                        await notif.notify_channel.send(f"{notif.member.mention}--{ping_member_data}")
+            await sleep(10)
```
Comment (Xithrius) [inline bot/cogs/rss.py:62]: I don't believe that `on_ready` is a good idea due to it having the chance of triggering multiple times. Here's an advanced startup example from the discord.py repo:

https://github.com/Rapptz/discord.py/blob/master/examples/advanced_startup.py#L32-L51

File: bot/cogs/rss.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    async def pull_rss_feed(self, url: str):
+        headers = {
+            "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) "
+                          "AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.135 Safari/537.36 Edge/12.246"
+        }
+        async with self.bot.session.get(url, headers=headers) as response:
+            return await response.text()
+
+    @hybrid_command(brief="Create or delete a new RSS notification", aliases=["create_notification", "notification"])
```
Comment (Xithrius) [inline bot/cogs/rss.py:72]: Aliases are constants.

```suggestion
    @hybrid_command(brief="Create or delete a new RSS notification", aliases=("create_notification", "notification"))
```

File: bot/cogs/rss.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+class NotifyMember:
+    def __init__(self, search: str, member: discord.Member, channel: discord.TextChannel):
+        self.search_term: str = search
+        self.member: discord.Member = member
+        self.notify_channel: discord.TextChannel = channel
+        self.notified: bool = False
+
+
+notifications: Dict[int, Dict[str, NotifyMember]] = {}
```
Comment (Xithrius) [inline bot/cogs/rss.py:24]: Perhaps this should be within the `RSS` cog.

File: bot/cogs/rss.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+
+def data_to_ping_from_rss(search_term: str, feed_data: List[dict]):
```
Comment (Xithrius) [inline bot/cogs/rss.py:27]: Add return annotation.

File: bot/cogs/rss.py (hunk 1 above)
Comment (Xithrius) [inline bot/cogs/rss.py:62]: Should the sleep here be a constant?

File: bot/cogs/rss.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                    if ping_member_data and not notif.notified:
+                        notif.notified = True
+                        await notif.notify_channel.send(f"{notif.member.mention}--{ping_member_data}")
+            await sleep(10)
+
+    async def pull_rss_feed(self, url: str):
+        headers = {
+            "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) "
+                          "AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.135 Safari/537.36 Edge/12.246"
+        }
```
Comment (Xithrius) [inline bot/cogs/rss.py:68]: Move out to constant.

File: bot/cogs/rss.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+            for user_id in notifications:
+                user_notifs = notifications[user_id]
+                for notif in user_notifs.values():
+                    ping_member_data = data_to_ping_from_rss(notif.search_term, rss_data)
+                    if ping_member_data and not notif.notified:
+                        notif.notified = True
+                        await notif.notify_channel.send(f"{notif.member.mention}--{ping_member_data}")
+            await sleep(10)
+
+    async def pull_rss_feed(self, url: str):
```
Comment (Xithrius) [inline bot/cogs/rss.py:64]:
```suggestion
    async def pull_rss_feed(self, url: str) -> str:
```

**Review by Xithrius — CHANGES_REQUESTED** [review]
A couple things.

File: bot/cogs/rss.py (hunk 1 above)
Comment (Xithrius) [reply bot/cogs/rss.py:62]: To be replaced with this instead: https://discordpy.readthedocs.io/en/stable/ext/tasks/index.html

File: bot/cogs/rss.py (hunk 3 above)
Comment (fisher60) [reply bot/cogs/rss.py:28]: That is probably preferred, but I think it is acceptable for now and I plan to refactor this cog fully in the near future so I will probably leave as-is.

Comment (Xithrius) [reply bot/cogs/rss.py:28]: Very well.

**Review by Xithrius — APPROVED** [review]
Everything has been resolved, I see no issues.

**Pull request merged**